### PR TITLE
first attempt at common code building under gradle

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>
- Abstracted common function between maven and gradle into a common layer.
- Changed gradle plugin to use this layer.
- Build common layer using gradle.
- All projects publish to the target maven repo so the docker image can pick them up.
- Tested with a test project

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-gradle-pr-33-v2tlq for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-gradle-pr-33-lh7r8 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

File: dev.galasa.plugin.common.impl/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (techcobweb) [inline dev.galasa.plugin.common.impl/pom.xml:1]: These pom.xmls all need deleting. Not needed.

Comment (techcobweb) [reply dev.galasa.plugin.common.impl/pom.xml]: Deleted.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: dev.galasa.plugin.common/build.gradle [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                        system = 'GitHub'
+                        url = 'https://github.com/galasa-dev/projectmanagement/issues'
+                    }
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            url  = "$targetMaven"
```
Comment (jadecarino) [inline dev.galasa.plugin.common/build.gradle:157]: Maybe I'm not understanding what this does with Maven Central, can we discuss?

File: README.md [hunk 3]
```diff
@@ -58,11 +122,24 @@ To use the Gradle OBR plugin in a Gradle test project:
...
     If you do this, ensure your `include` statement in your root project's `settings.gradle` file matches the name given to your OBR subproject.
 
-3. Run `gradle build` to build the OBR.
+5. Building artifacts
+
+    To verify that the OBR was successfully built, A `gradle build` directory will appear in your OBR directory and within it, a `galasa.obr` file should be present.
 
-    To verify that the OBR was successfully built, A `build` directory will appear in your OBR directory and within it, a `galasa.obr` file should be present.
+6. Publishing artifacts to the local maven repository
+    If you run `gradle clean build publishMaven` then the artifacts built will be published to the local maven folders on the build machine.
```
Comment (jadecarino) [inline README.md:130]: Is this not publishToMavenLocal?

File: dev.galasa.gradle.impl/src/main/java/dev/galasa/gradle/testcatalog/ErrorRaiserGradleImpl.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @Override
+    public void raiseError(String template, Object... params) throws TestCatalogException {
+        String msg = MessageFormat.format(template,params);
+        log.error(msg);
+        throw new TestCatalogException(msg);
+    }
+
+    @Override
+    public void raiseError(Throwable cause, String template, Object... params) throws TestCatalogException {
+        String msg = MessageFormat.format(template,params)+" cause:"+cause.toString();
```
[the remark this answers is not in the document]
Comment (jadecarino) [reply dev.galasa.gradle.impl/src/main/java/dev/galasa/gradle/testcatalog/ErrorRaiserGradleImpl.java:31]: I mean the actual line outputted by the logger might be hard to read, not the line of code :)

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: dev.galasa.gradle.impl/src/main/java/dev/galasa/gradle/testcatalog/ErrorRaiserGradleImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (techcobweb) [reply dev.galasa.gradle.impl/src/main/java/dev/galasa/gradle/testcatalog/ErrorRaiserGradleImpl.java:31]: The templates for the messages contain spaces. I added an extra space after the 'cause:' to make it easier as you suggest.

File: dev.galasa.gradle.impl/src/main/java/dev/galasa/gradle/testcatalog/TestCatalogDeployTask.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            throw new TestCatalogException("The test stream is missing in the deployTestCatalog extension");
+        }
+        
+        String testStream = extension.stream.trim().toLowerCase();
+
+        Path pathArtifact = Paths.get(getInputs().getFiles().getSingleFile().toURI());
+
+        String galasaAccessToken = extension.getGalasaToken();
+
+        // Instantiate all the classes to isolave gradle dependencies...
```
[the remark this answers is not in the document]
Comment (techcobweb) [reply dev.galasa.gradle.impl/src/main/java/dev/galasa/gradle/testcatalog/TestCatalogDeployTask.java:47]: fixed

File: dev.galasa.plugin.common.impl/build.gradle [hunk 6]
```diff
@@ -0,0 +1,139 @@
+plugins {
+    id 'java'
+    id 'jacoco'
+    id 'maven-publish'
+    id 'signing'
+}
+
+group   = "dev.galasa"
+version = "0.32.0"
```
[the remark this answers is not in the document]
Comment (techcobweb) [reply dev.galasa.plugin.common.impl/build.gradle]: Extracted to a symbolic constant, and commentsed.

[the remark this answers is not in the document]
Comment (techcobweb) [reply dev.galasa.plugin.common.impl/build.gradle]: I uplifted to 0.33.0 everywhere.

File: dev.galasa.plugin.common.impl/src/main/java/dev/galasa/plugin/common/impl/auth/AuthenticationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    
+
+    private HttpPost createAuthHttpPostRequest(Gson gson, String authEndpointUrl) throws AuthenticationException {
+
+        // Create the post request.
+        HttpPost postRequest = new HttpPost(authEndpointUrl);
+        
+        // Set headers..
+        postRequest.setHeader("Content-Type","application/json");
+        postRequest.setHeader("ClientApiVersion","0.32.0");
```
[the remark this answers is not in the document]
Comment (techcobweb) [reply dev.galasa.plugin.common.impl/src/main/java/dev/galasa/plugin/common/impl/auth/AuthenticationServiceImpl.java]: Extracted to a symbolic constant and explained why we don't need to bump it up every time.

File: dev.galasa.plugin.common/build.gradle (hunk 2 above)
Comment (techcobweb) [reply dev.galasa.plugin.common/build.gradle:157]: Discussed on a call. Fixed by removing gradle properties which default the targetMaven value erroneously.

File: README.md (hunk 3 above)
Comment (techcobweb) [reply README.md:130]: Fixed

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-gradle-pr-33-fr4q8 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Review by jadecarino — APPROVED** [review]

**Pull request merged**